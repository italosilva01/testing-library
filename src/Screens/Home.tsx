import React from "react";

import smallLogo from "../assets/logo-pequeno.png";
import largeLogo from "../assets/logo.png";
import participante from "../assets/participante.png";
import { Form } from "components/Form/Form";
import { Card } from "components/Card/Card";
import { Foot } from "components/Foot/Foot";
export const Home = () => {
  return (
    <div className="flex flex-col h-screen bg-blue-royal ">
      <header className="flex flex-col bg-blue-royal items-center md:px-24 md:flex-row md:justify-between">
        <img
          src={smallLogo}
          alt="Logo Sorteador de Amigo secreto"
          width={186}
          height={157}
          className="block md:hidden"
        />
        <img
          src={largeLogo}
          alt="Logo Sorteador de Amigo secreto"
          width={186}
          height={157}
          className="hidden md:block"
        />
        <img
          src={participante}
          alt="Desenho de um participante"
          width={328}
          height={158}
        />
      </header>

      <Card>
        <Form />
      </Card>
      <Foot />
    </div>
  );
};
