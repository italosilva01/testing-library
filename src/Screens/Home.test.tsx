import { render } from "@testing-library/react";
import { RecoilRoot } from "recoil";
import { Home } from "./Home";

const mockNavigate = jest.fn();
jest.mock("react-router-dom", () => {
  return {
    useNavigate: () => mockNavigate,
  };
});

describe("The settings' page", () => {
  test("The page should be rendered correctly", () => {
    const { container } = render(
      <RecoilRoot>
        <Home />
      </RecoilRoot>
    );

    expect(container).toMatchSnapshot();
  });
});
