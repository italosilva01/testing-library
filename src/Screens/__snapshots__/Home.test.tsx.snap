// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The settings' page The page should be rendered correctly 1`] = `
<div>
  <div
    class="flex flex-col h-screen bg-blue-royal"
  >
    <header
      class="flex flex-col bg-blue-royal items-center md:px-24 md:flex-row md:justify-between"
    >
      <img
        alt="Logo Sorteador de Amigo secreto"
        class="block md:hidden"
        height="157"
        src="logo-pequeno.png"
        width="186"
      />
      <img
        alt="Logo Sorteador de Amigo secreto"
        class="hidden md:block"
        height="157"
        src="logo.png"
        width="186"
      />
      <img
        alt="Desenho de um participante"
        height="158"
        src="participante.png"
        width="328"
      />
    </header>
    <div
      class=" flex flex-1 bg-white-cream p-20 mt-n16 justify-center rounded-t-64px border-2 border-black"
      role="container"
    >
      <form>
        <input
          placeholder="Insira os nomes dos participantes"
          value=""
        />
        <button
          disabled=""
          type="submit"
        >
          Adicionar
        </button>
        
      </form>
    </div>
    <div>
      <button
        disabled=""
      >
        Iniciar brincadeira!
      </button>
    </div>
  </div>
</div>
`;
